File: C++/binary_search.cpp
#include<iostream>
#include<string>
#include<climits>
using namespace std;

int search(vector<int>& nums, int target) {
        int left=0,right=nums.size()-1;
        while (left<=right){
            int mid=(left+right)/2;
            if(mid==target){
                return mid;
            }
            else if (mid>target){
                right=mid-1;
            }
            else{
                left=mid+1;
            }
        }
        return -1;
    }

int main() {
    vector<int> numbers = {2,7,11,15};
    int target = 9;
    int result= search(numbers, target);

    cout<< result<< endl;
    return 0;
}

File: Array/BinarySearch.java
public class BinarySearch{
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Check if target is present at mid
            if (arr[mid] == target) {
                return mid; // Found
            }
            // If target greater, ignore left half
            else if (arr[mid] < target) {
                left = mid + 1;
            }
            // If target smaller, ignore right half
            else {
                right = mid - 1;
            }
        }

        return -1; // Not found
    }

    public static void main(String[] args) {
        int[] numbers = {2, 4, 6, 8, 10, 12, 14};
        int target = 10;

        int result = binarySearch(numbers, target);

        if (result != -1)
            System.out.println("Element found at index: " + result);
        else
            System.out.println("Element not found in array.");
    }
}
